bastion.cfn.yml

```diff
@@ -16,68 +16,290 @@ Parameters:
 
   KeyName:
     Description: EC2 key pair name for bastion host SSH access
+    Type: AWS::EC2::KeyPair::KeyName
+
+  LogRetentionInDays:
+    Description: Number of days you would like you cloudwatch logs to be retained
+    Type: Number
+    Default: 7
+
+  MFA:
+    Description: Change it to true to install MFA using Google Authenticator on your instance
     Type: String
-    MinLength: 1
-    MaxLength: 255
-    AllowedPattern: "[\\x20-\\x7E]*"
-    ConstraintDescription: Key pair name can contain only ASCII characters.
+    ConstraintDescription: Value must be true or false
+    Default: false
+    AllowedValues:
+      - true
+      - false
 
 Mappings:
 
   AMIMap:  # Amazon Linux AMI
     ap-northeast-1:
-      AMI: ami-4af5022c
+      AMI: ami-1a15c77b
     ap-northeast-2:
-      AMI: ami-8663bae8
+      AMI: ami-a04297ce
     ap-south-1:
-      AMI: ami-d7abd1b8
+      AMI: ami-cacbbea5
     ap-southeast-1:
-      AMI: ami-fdb8229e
+      AMI: ami-7243e611
     ap-southeast-2:
-      AMI: ami-30041c53
+      AMI: ami-55d4e436
     eu-west-1:
-      AMI: ami-ebd02392
-    eu-west-2:
-      AMI: ami-489f8e2c
+      AMI: ami-d41d58a7
     eu-central-1:
-      AMI: ami-657bd20a
+      AMI: ami-0044b96f
     sa-east-1:
-      AMI: ami-d27203be
+      AMI: ami-b777e4db
     us-east-1:
-      AMI: ami-4fffc834
-    us-east-2:
-      AMI:  ami-ea87a78f
+      AMI: ami-c481fad3
     us-west-1:
-      AMI: ami-3a674d5a
+      AMI: ami-de347abe
     us-west-2:
-      AMI: ami-aa5ebdd2
-    ca-central-1:
-      AMI: ami-5ac17f3e
+      AMI: ami-b04e92d0
 
 Resources:
 
+  LogRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+        - Effect: Allow
+          Principal:
+            Service:
+            - ec2.amazonaws.com
```

Comment:
Nit: You should be able to pass these as straight values rather than collections of one item.